nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/QueryRowsMarkLogic.java

```diff
@@ -81,10 +81,18 @@ public void onTrigger(ProcessContext context, ProcessSessionFactory sessionFacto
 			PlanBuilder.Plan plan = rowManager.newRawPlanDefinition(new StringHandle(jsonPlan));
 			InputStream inputStream = rowManager.resultDoc(plan, new InputStreamHandle().withMimetype(mimeType)).get();
 
-			if (inputStream != null) {
-				flowFile = session.write(flowFile, out -> {
-					IOUtils.copy(inputStream, out);
-				});
+			InputStreamHandle handle = new InputStreamHandle();
+			if( handle != null) {
```

Comment:
I think the only thing we need here is a try/finally that closes inputStream, right? In this code, handle will never be null, and handle doesn't need to be closed either. We just need to close inputStream after trying to write it. I'll submit a separate fix.

Reply:
Good point! Following the original too closely :) I went ahead and made the change you suggested